src/components/calcite-input/calcite-input.tsx

```diff
@@ -89,6 +89,8 @@ export class CalciteInput {
   /** optionally add prefix  **/
   @Prop() prefixText?: string;
 
+  @Prop({ reflect: true }) uneditable = false;
```

Comment:
Why not just use the `readonly` feature of html input?  Seems like the use case for this could work with just that attribute.

Reply:
Another option I just thought of is you could just hide the slotted `calcite-input` inside the `inline-editable`'s css with something like
```
/* calcite-inline-editable.scss */
:host([uneditable]) {
  ::slotted(calcite-input) {
    display: none;
  }
}
```
That way you don't need to add support for making the input readonly at all and can just display the text like normal since you're storing the value in `this.valuePriorToEditing`.

Reply:
I first tried to name my new prop readonly, but there's already conflicting readonly styles baked into the calcite-input component to handle that attribute. I introduced `uneditable` to get around those. In the example above, are you proposing that some other element would house the value for display prior to showing the input? This is the pattern I followed in my previous PR (#1158), but that required a bunch of extra styles to match the sizing/spacing of the input. I previously tried using a calcite-label and then swapping it out for the calcite-input. That mostly worked, but it introduced a problem when trying to wrap a calcite-input with a calcite-label. Thought it was simpler to use the calcite input and just add a new style variant to accommodate this uneditable/readonly state. The one gotcha around this approach is that I need to disable (without styling the input as disabled) the input when it is uneditable so that a user can't tab into the input when it is in the readonly state and then start typing away.

Reply:
@bstifle should the `readonly` styles already baked into `calcite-input` also apply to when input is used inside `inline-editable`?

Reply:
> In the example above, are you proposing that some other element would house the value for display prior to showing the input? This is the pattern I followed in my previous PR (#1158), but that required a bunch of extra styles to match the sizing/spacing of the input. I previously tried using a calcite-label and then swapping it out for the calcite-input. That mostly worked, but it introduced a problem when trying to wrap a calcite-input with a calcite-label. Thought it was simpler to use the calcite input and just add a new style variant to accommodate this uneditable/readonly state. The one gotcha around this approach is that I need to disable (without styling the input as disabled) the input when it is uneditable so that a user can't tab into the input when it is in the readonly state and then start typing away.

It fees like `calcite-input` should look consistent when it is `readonly` and/or `disabled` whether it is used standalone or inside something like `inline-editable`.  And since there are already readonly and disabled styles built into `calcite-input`, those should be taken advantage of for this component without the need to establish another prop that has an ambiguous meaning to "readonly" and "disabled".

I think if we get buy-in from @bstifle on this, that we could do away with the `uneditable` prop and keep the implementation as you have it without managing extra state to house the value when the input is disabled.

Reply:
I think the idea is that these don't look like readonly / disabled inputs, they appear as plain text, so I think that visual distinction is actually good to have for "inline editing" kind of workflows - its a little different design need. Ideally the uneditable prop isn't needed though, agree on that.

Reply:
The "readonly / disabled" functions the same in this state, since input uses "scoped" vs. shadow, those can actually be overridden from this component and (I think) that would work here.

Reply:
> I think the idea is that these don't look like readonly / disabled inputs, they appear as plain text

This idea would lend me to support rendering the value of the input as a `span` when not in edit mode, which would mean either hiding or not rendering the input at all, in either case obviating the need to have the `uneditable` prop or explicitly disabling/readonly-ing the input.  Ideally there shouldn't need to be any updates to `calcite-input` for this new component, it would just get used like it normally does since it wouldn't be visible unless the value is being edited.